### PR TITLE
Remove/replace all UFE_PREVIEW_VERSION_NUM

### PR DESCRIPTION
**Pull request by samuelliu-adsk**
Remove/replace all UFE_PREVIEW_VERSION_NUM for that latest Maya has UFE v4

### CODE REVIEW

File: lib/mayaUsd/ufe/Utils.cpp [hunk 1]
```diff
@@ -962,41 +962,39 @@ namespace {
...
+        { SdfValueTypeNames->Matrix4d.GetHash(), Ufe::Attribute::kMatrix4d },   // GfMatrix4d
 #endif
     };
     const auto iter = sUsdTypeToUfe.find(usdType.GetHash());
     if (iter != sUsdTypeToUfe.end()) {
         return iter->second;
     } else {
-        static const std::unordered_map<std::string, Ufe::Attribute::Type> sCPPTypeToUfe
-        {
+        static const std::unordered_map<std::string, Ufe::Attribute::Type> sCPPTypeToUfe {
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/Utils.cpp:991]: clang-format change?

File: lib/mayaUsd/ufe/Utils.cpp [hunk 2]
```diff
@@ -1098,27 +1096,26 @@ Ufe::Attribute::Type usdTypeToUfe(const PXR_NS::UsdAttribute& usdAttr)
 SdfValueTypeName ufeTypeToUsd(const Ufe::Attribute::Type ufeType)
 {
     // Map the USD type into UFE type.
-    static const std::unordered_map<Ufe::Attribute::Type, SdfValueTypeName> sUfeTypeToUsd
-    {
+    static const std::unordered_map<Ufe::Attribute::Type, SdfValueTypeName> sUfeTypeToUsd {
```
Comment (seando-adsk) [inline lib/mayaUsd/ufe/Utils.cpp:1099]: clang-format?

File: test/lib/ufe/testShaderNodeDef.py [hunk 3]
```diff
@@ -93,8 +94,8 @@ def testDefinitions(self):
         for mtlxDef in mtlxDefs:
             self.assertTrue(mtlxDef in nodeDefTypes)
 
-    @unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') < '4001', 'nodeDefHandler is only available in UFE preview version 0.4.1 and greater')
-    @unittest.skipIf(os.getenv('UFE_PREVIEW_VERSION_NUM', '0000') >= '4015', 'filename and float2 are no longer generic starting with Ufe 0.4.15')
+    @unittest.skipUnless(ufeUtils.ufeFeatureSetVersion() >= 4, 'nodeDefHandler is only available in UFE v4 or greater')
+    @unittest.skipIf(ufeUtils.ufeFeatureSetVersion() >= 4, 'filename and float2 are no longer generic starting with Ufe v4')
```
Comment (seando-adsk) [inline test/lib/ufe/testShaderNodeDef.py:98]: Something seems off here. I think we might just need the test below (line 143). See original PR where this was split https://github.com/Autodesk/maya-usd/commit/4f34c98259da09542763822103f5f525fa985c0d#diff-704929d4dbdc0ef77e39bdc75501acb1e470f82877cad0f5eaab20a27823e35b

**Review by seando-adsk — CHANGES_REQUESTED** [review]
Good work, thanks. Just a few minor things to fixup.
Can you also add Maya 2023 at 132 here: https://github.com/Autodesk/maya-usd/blob/dev/test/testUtils/mayaUtils.py#L281

File: lib/mayaUsd/ufe/Utils.cpp [hunk 4]
```diff
@@ -962,41 +962,39 @@ namespace {
...
-            { SdfValueTypeNames->Color3d.GetHash(), Ufe::Attribute::kColorFloat3 }, // GfVec3d
-#if (UFE_PREVIEW_VERSION_NUM >= 4015)
-            { SdfValueTypeNames->Asset.GetHash(), Ufe::Attribute::kFilename },      // SdfAssetPath
-            { SdfValueTypeNames->Float2.GetHash(), Ufe::Attribute::kFloat2 },       // GfVec2f
-            { SdfValueTypeNames->Float4.GetHash(), Ufe::Attribute::kFloat4 },       // GfVec4f
-            { SdfValueTypeNames->Color4f.GetHash(), Ufe::Attribute::kColorFloat4 }, // GfVec4f
-            { SdfValueTypeNames->Color4d.GetHash(), Ufe::Attribute::kColorFloat4 }, // GfVec4d
-            { SdfValueTypeNames->Matrix3d.GetHash(), Ufe::Attribute::kMatrix3d },   // GfMatrix3d
-            { SdfValueTypeNames->Matrix4d.GetHash(), Ufe::Attribute::kMatrix4d },   // GfMatrix4d
+    static const std::unordered_map<size_t, Ufe::Attribute::Type> sUsdTypeToUfe {
```
[the remark this answers is not in the document]
Comment (samuelliu-adsk) [reply lib/mayaUsd/ufe/Utils.cpp:965]: Hi Sean,

Yes I'm using clang-format version 10.0.0 and somehow it got formatted this way. I checked again and still got the same result. It is passing the format checking preflight tho, does that mean it is correct?

File: test/lib/ufe/testShaderNodeDef.py (hunk 3 above)
Comment (samuelliu-adsk) [reply test/lib/ufe/testShaderNodeDef.py:98]: Yes my bad. The testDefinitionByType test is only for 4001 to 4015. So we only need to run testDefinitionByType.

File: lib/mayaUsd/ufe/Utils.cpp (hunk 1 above)
Comment (samuelliu-adsk) [reply lib/mayaUsd/ufe/Utils.cpp:991]: Same here. I think the change is valid. Seems like a chain effect of my change, clang formatted it this way not..

**Review by seando-adsk — APPROVED** [review]
Everything looks good. Thanks.

**Review by seando-adsk — APPROVED** [review]

**Pull request merged**